lib/ecl/ecl_smspec.c

```diff
@@ -364,9 +366,36 @@ static void ecl_smspec_fortio_fwrite( const ecl_smspec_type * smspec , fortio_ty
   int num_nodes           = ecl_smspec_num_nodes( smspec );
   {
     ecl_kw_type * restart_kw = ecl_kw_alloc( RESTART_KW , SUMMARY_RESTART_SIZE , ECL_CHAR );
-    int i;
-    for (i=0; i < SUMMARY_RESTART_SIZE; i++)
-      ecl_kw_iset_string8( restart_kw , i , "");
+    const char * restart_case = smspec->write_restart_case;
+    if (restart_case != NULL) {
+       int restart_case_len = strlen(restart_case);
+       if (restart_case_len > 64)
+          util_abort("%s: Restart case name, %s, is too long w/ %d chars.\n", __func__, restart_case, restart_case_len);
+       else {                     
+          for (int i = 0; i < SUMMARY_RESTART_SIZE; i++) {
+             char str[9];
+             
+             for (int j = 0; j < 9; j++) {
+                int index = 8 * i + j;
```

Comment:
Unsure as to whether the file format requires restart_kw to be of size SUMMARY_RESTART_SIZE or not. 
Can it be 1 ?